### ISSUE TITLE
Compiler error produces output that breaks terminal rendering

### ISSUE THREAD

**Issue by hovsater** [elm/compiler#2302]
**Quick Summary:**

While working on https://github.com/hovsater/aoc-elm I encountered an error message that breaks rendering inside the terminal. I have no idea what it is but have been able to re-produce it on three different terminals as well as having others from the Elm community confirm these findings.

```console
$ git clone -b weird-error git@github.com:hovsater/aoc-elm
$ elm make --output=/dev/null src/Main.elm
```

Observe the error generated by the compiler. There's large amount of whitespace (non-visible characters) between the first line and the last line. If you copy the full error message and paste it elsewhere, this isn't visible.

## SSCCE

- **Elm:** 0.19.1
- **Browser:** None
- **Operating System:** macOS Ventura 13.3

**Comment by lydell** [comment]
As far as I can tell, “breaks terminal rendering” is a bit of a red herring. What seems to happen is that line 6 of the error message contains 65532 spaces in a row. Terminals wrap long lines, which in effect results in a very long empty section to scroll through in the middle of the error message.

Here I replaced long sequences of a character like `XXXXXXX…` with `{X}*42`:

```
❯ elm make Main.elm 2> output.txt
Detected problems in 1 module.

❯ node -p 'fs.readFileSync("output.txt", "utf8").replace(/(.)\\1{5,}/g, m => `{${m[0]}}*${m.length}`)'
-- NAME CLASH {-}*45 ui-explorer/Main.elm

This `case` pattern has multiple `a` variables.

6|{ }*9[ a, a ] -> list
{ }*16^{ }*65532^
How can I know which one you want? Rename one of them!
```

Random note: 65532 is suspiciously close to 2^16 = 65,536.

**Comment by hovsater** [comment]
> As far as I can tell, “breaks terminal rendering” is a bit of a red herring.

You're right. That was probably some unfortunate wording on my part. I've updated the description accordingly. 🙂

**Comment by evancz** [comment]
Perhaps related to https://github.com/elm/compiler/issues/2334, perhaps something about JSON row/col "going negative" and ending up looking like 2^16. The row/col size in 0.19.2 is bumped up to 2^32, but I suspect the root issue here will remain the same.
